stdlib/public/runtime/CrashHandlerMacOS.cpp

```diff
@@ -231,16 +231,24 @@ handle_fatal_signal(int signum,
   void *pc = 0;
   ucontext_t *ctx = (ucontext_t *)uctx;
 
-#ifdef __DARWIN_UNIX03
+#if defined(__arm64__) && __DARWIN_OPAQUE_ARM_THREAD_STATE64
+#define THREAD_STATE_MEMBER(x) __opaque_##x
+#elif __DARWIN_UNIX03
+#define THREAD_STATE_MEMBER(x) __##x
+#else
+#define THREAD_STATE_MEMBER(x) x
+#endif
+
+#if __DARWIN_UNIX03
 #define CTX_MEMBER(x) __##x
 #else
 #define CTX_MEMBER(x) x
 #endif
 
 #if defined(__x86_64__)
-  pc = (void *)(ctx->uc_mcontext->CTX_MEMBER(ss).CTX_MEMBER(rip));
+  pc = (void *)(ctx->uc_mcontext->CTX_MEMBER(ss).THREAD_STATE_MEMBER(rip));
 #elif defined(__arm64__)
-  pc = (void *)(ctx->uc_mcontext->CTX_MEMBER(ss).CTX_MEMBER(pc));
+  pc = (void *)(ctx->uc_mcontext->CTX_MEMBER(ss).THREAD_STATE_MEMBER(pc));
```

Comment:
Would be good to strip this, since the signature bits won't be useful to us.

Reply:
It's really only a "we're doing a backtrace, here's the program counter in case it goes wrong" message, so I'm not too worried about it — we'll normally present it stripped very shortly afterwards anyway.

Reply:
In fact, thinking about it, it might be nice to keep the ptrauth bits here, because if we're interrupted while attempting to generate the backtrace, we aren't going to show a register dump (so we won't have any way to see what they were if we strip them, and if the failure happened for ptrauth reasons, we'd have no clue at that point).

Reply:
Gotcha, that all makes sense to me.